src/engine/plugins/nearest.cpp

```diff
@@ -26,7 +26,7 @@ Status NearestPlugin::HandleRequest(const api::NearestParameters &params,
 {
     BOOST_ASSERT(params.IsValid());
 
-    if (params.number_of_results > max_results)
+    if (static_cast<std::int64_t>(params.number_of_results) > max_results)
```

Comment:
Fixed.

We also should check other casts, ticketing over at https://github.com/Project-OSRM/osrm-backend/issues/2916